src/api/app/models/workflow/step/branch_package_step.rb

```diff
@@ -70,25 +69,29 @@ def check_source_access
         options = { use_source: false, follow_project_links: true, follow_multibuild: true }
         src_package = Package.get_by_project_and_name(source_project_name, source_package_name, options)
 
-        raise Pundit::NotAuthorizedError unless PackagePolicy.new(@token.user, src_package).create_branch?
+        raise Token::Errors::CanNotBranchPackageNoPermission, 'Your package could not be branched: not enough permissions.' unless
+          PackagePolicy.new(@token.user, src_package).create_branch?
       end
 
       def branch
         check_source_access
-        BranchPackage.new({ project: source_project_name, package: source_package_name,
-                            target_project: target_project_name,
-                            target_package: target_package_name }).branch
+
+        begin
+          BranchPackage.new({ project: source_project_name, package: source_package_name,
+                              target_project: target_project_name,
+                              target_package: target_package_name }).branch
+        rescue BranchPackage::InvalidArgument, InvalidProjectNameError, ArgumentError => e
+          raise Token::Errors::CanNotBranchPackage, "Your package could not be branched: #{e.message}"
+        rescue Project::WritePermissionError, CreateProjectNoPermission => e
+          raise Token::Errors::CanNotBranchPackageNoPermission, "Your package could not be branched, not enough permissions: #{e.message}"
```

Comment:
To make the error message a bit clearer in case there are multiple branch package steps, I suggest this:

```suggestion
          raise Token::Errors::CanNotBranchPackageNoPermission, "Package #{source_project_name}/#{source_package_name} could not be branched due to missing permissions: #{e.message}"
```